### PR TITLE
Issue#354 Test cases added for Gradle and Maven

### LINKED ISSUE #354: Test Case for Handling Spaces in App name  and Directory Name.
Include test case for the following:
The application name has a space.
The directory name contains a space.

### PR DESCRIPTION
**Pull request by SuparnaSuresh**
Fixes https://github.com/OpenLiberty/liberty-tools-vscode/issues/354
Added test case to handling the scenario for 

- Gradle project name with space 
- Gradle Project directory with space
- Maven Project directory with space

### CODE REVIEW

File: src/test/GradleTestDevModeActionsWSProjectNameInit.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import {  WebDriver, VSBrowser } from 'vscode-extension-tester';
+import * as utils from './utils/testUtils';
+
+describe('Open Gradle Project - Project name with space', () => {
+
+    let driver: WebDriver;  
+
+    before(() => {    
+        utils.renameProject();
```
Comment (aparnamichael) [inline src/test/GradleTestDevModeActionsWSProjectNameInit.ts:9]: Please add comments, that will be helpful to understand about the newly added methods.

File: src/test/GradleTestDevModeActionsWSProjectNameTasks.ts [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+   section = await contentPart.getSection('Liberty Dashboard');   
+   console.log("Found Liberty Dashboard....");
+   expect(section).not.undefined; 
+ 
+}).timeout(25000);
+
+
+it('Open dasboard shows items - Gradle', async () => {
+
+
```
Comment (aparnamichael) [inline src/test/GradleTestDevModeActionsWSProjectNameTasks.ts:31]: Please format the file and remove unwanted spaces.

File: src/test/GradleTestPathDevModeActionsWSTasks.ts [hunk 3]
```diff
@@ -0,0 +1,265 @@
+import { expect } from 'chai';
+import { InputBox, Workbench,SideBarView, ViewItem, ViewSection,EditorView,DefaultTreeItem, DebugView } from 'vscode-extension-tester';
+import * as utils from './utils/testUtils';
+import * as constants from './definitions/constants';
+import path = require('path');
+// import { after } from 'mocha';
```
Comment (aparnamichael) [inline src/test/GradleTestPathDevModeActionsWSTasks.ts:6]: remove unused/commented line of codes.

File: src/test/GradleTestDevModeActionsWSProjectNameInit.ts (hunk 1 above)
Comment (SuparnaSuresh) [reply src/test/GradleTestDevModeActionsWSProjectNameInit.ts]: Comments added for new methods. Thanks.

File: src/test/GradleTestDevModeActionsWSProjectNameTasks.ts (hunk 2 above)
Comment (SuparnaSuresh) [reply src/test/GradleTestDevModeActionsWSProjectNameTasks.ts]: Formatted file and removed unwanted spaces. Thanks

File: src/test/GradleTestPathDevModeActionsWSTasks.ts (hunk 3 above)
Comment (SuparnaSuresh) [reply src/test/GradleTestPathDevModeActionsWSTasks.ts]: Removed commented lines. Thanks.

**Comment by SuparnaSuresh** [comment]
@aparnamichael 
Review comments addressed. 
Made changes to run tests for 'Gradle project name' with space and 'Gradle Project directory with space' together.

**Review by aparnamichael — DISMISSED** [review]
Looks good!!

**Review by TrevCraw — CHANGES_REQUESTED** [review]
It looks like the test suite for dev actions was copied entirely for both maven and gradle to run against some different projects. If a change to a test is needed, we would need to make the change in multiple places. This will lead to difficult code management, especially if we continue to copy tests anytime we need to test against a different kind of project (think multi-module projects). 

I believe the ultimate goal here is to run the existing dev actions tests against projects with spaces in the path. We should not copy the tests. We should find a way to run the same tests against various projects (i.e. find a way to parameterize the projects for the tests).

**Comment by SuparnaSuresh** [comment]
Hi @TrevCraw,
I have analyzed the approach to running the same tests across different projects. It is possible to implement the same test cases for multiple projects, but this requires restructuring the current setup. Specifically, since the VSCode Extension Tester executes test cases sequentially based on filenames, the entire test case structure needs to be updated.

Currently, test cases are split across two files—init and devActions. 
Gradle project --> We have GradleInitProject.ts (Open the IDE and project) and GradleTestDevModeActions.ts (executing the dev actions)
Maven project --> MavenInitProject.ts (Open IDE and project) and MavenTestDevModeActions.ts (cover dev action test cases)
To enable reusability across different projects, all test cases will need to be consolidated into a single file. Additionally, we will need to update the relevant methods in testUtils.ts to reflect these changes.

In summary, we will need to completely restructure the current test cases. This is expected to take approximately 3-4 days of work.
Kindly confirm whether can we proceed with the approach.
Thanks

**Comment by aparnamichael** [comment]
@SuparnaSuresh - Please update the branch and resolve the conflicts.
